torchquad/integration/base_integrator.py

```diff
@@ -77,8 +79,15 @@ def evaluate_integrand(fn, points, weights=None,args=None):
             )
 
         if weights is not None:
+            if len(result.shape) > 1:
```

Comment:
Could you add a comment here explain this line? I think this is related to the changes from #160 ?